gui/src/main/java/kcl/teamIndexZero/traffic/gui/GUI_Primitives.java

```diff
@@ -0,0 +1,138 @@
+package kcl.teamIndexZero.traffic.gui;
+
+import com.sun.org.apache.xml.internal.resolver.helpers.PublicId;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.geom.AffineTransform;
+import java.awt.geom.QuadCurve2D;
+import java.io.*;
+
+/**
+ * Created by kumar awijeet on 2/24/2016. thanks for Working!!
+ */
+public class GUI_Primitives {
+    public void drawLine(int x1, int y1, int x2, int y2, Graphics g) {
+        Graphics2D g2 = (Graphics2D) g;
+        g.drawLine(x1, y1, x2, y2);
+    }
+
+    public void drawCurve(int x1, int y1, int x2, int y2, int ctrlx, int ctrly, Graphics g) {
+        Graphics2D g2 = (Graphics2D) g;
+        QuadCurve2D q = new QuadCurve2D.Float();
+        q.setCurve(x1, y1, ctrlx, ctrly, x2, y2);
+        g2.draw(q);
+    }
+
+    public void drawSingleRoad(int x, int y, int x1, int y1, int x2, int y2, int x3, int y3, Graphics g) {
+        drawLine(x, y, x1, y1, g);
+        drawLine(x2, y2, x3, y3, g);
+    }
+    public void drawTruck(int x3, int y3, double angleToXAxis, Graphics g)
+    {
+        //files taken from http://hdimagelib.com/trailer+truck+top+view
```

Comment:
I will try to find that, but if it's not there, can we still use it? There is a © on the second picture but other than that found nothing as to what you are asking.                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the EE network.                                                                                                                                                                                                                From: EsSent: Monday 21 March 2016 01:11To: teamIndexZero/index_zero_trafficsystemReply To: teamIndexZero/index_zero_trafficsystemCc: kumar awijeetSubject: Re: [index_zero_trafficsystem] Gui graphics kumar awijeet (#9)In gui/src/main/java/kcl/teamIndexZero/traffic/gui/GUI_Primitives.java:
> +    }
> +
> +    public void drawCurve(int x1, int y1, int x2, int y2, int ctrlx, int ctrly, Graphics g) {
> +        Graphics2D g2 = (Graphics2D) g;
> +        QuadCurve2D q = new QuadCurve2D.Float();
> +        q.setCurve(x1, y1, ctrlx, ctrly, x2, y2);
> +        g2.draw(q);
> +    }
> +
> +    public void drawSingleRoad(int x, int y, int x1, int y1, int x2, int y2, int x3, int y3, Graphics g) {
> +        drawLine(x, y, x1, y1, g);
> +        drawLine(x2, y2, x3, y3, g);
> +    }
> +    public void drawTruck(int x3, int y3, double angleToXAxis, Graphics g)
> +    {
> +        //files taken from http://hdimagelib.com/trailer+truck+top+view

The image must have a proper license that allows at least academic use. Please check and put the relevant information in the report.

—You are receiving this because you authored the thread.Reply to this email directly or view it on GitHub